src/transaction/page_broker.cpp.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "page_broker.hpp"  // for convenience - code completion.
+
+template <typename PageT>
+entry_state
+page_broker<PageT>::register_entry (typename map_type<PageT>::key id)
+{
+  std::unique_lock<std::mutex> lock (m_pages_mutex);
+
+  entry_state result = EXISTING_ENTRY;
+  if (m_requested_page_id_count.find (id) != m_requested_page_id_count.end ())
+    {
+      m_requested_page_id_count[id]++;
+    }
+  else
```

Comment:
As @cristiarg pointed out too, when ID is found, it is actually searched twice. Once by find() and once by operator[]. Save the iterator from find() to avoid the double lookup.